.github/workflows/doc_gen.yml

```diff
@@ -0,0 +1,40 @@
+name: Check document generation
+
+on:
+  push:
+    branches:
+    - asf-site
+  pull_request:
+    branches:
+    - asf-site
+  workflow_dispatch:
+    inputs:
+      target:
+        description: 'Target branch to run'
+        required: true
```

Comment:
I think you can remove this `workflow_dispatch`. It's pretty unlikely that people manually trigger the GA build ..

Reply:
Thanks! Removed. To keep the commit order I have modified the first commit and force pushed the commits.